controllers/bpfman-agent/common.go

```diff
@@ -633,6 +633,31 @@ func (r *ReconcilerCommon) handleProgDelete(
 	return internal.Unchanged, nil
 }
 
+func (r *ReconcilerCommon) unloadAndDeleteApplicationProgramsList(ctx context.Context, bpfDeletedPrograms *bpfmaniov1alpha1.BpfProgramList) error {
```

Comment:
```suggestion
func (r *ReconcilerCommon) unloadAndDeleteBpfPrograms(ctx context.Context, bpfProgramList *bpfmaniov1alpha1.BpfProgramList) error {
```
This can also be used by handleProgCreateOrUpdate() to delete the extra BpfPrograms